### PR TITLE
idempotent license report

### PR DESCRIPTION
**Pull request by SylvainJuge**
When updating the license report with `./gradlew generateLicenseReport`, the generated license report contains the current date.

There is unfortunately no option to disable it in the gradle plugin used to generate the report https://github.com/jk1/Gradle-License-Report.

This PR filters the content of the generated report and removes the date using a known pattern regex.

Doing this means that the `generateLicenseReport` could be executed systematically to keep the license-report up-to-date when changes are made to dependencies, for example using a git pre-commit or pre-push hook.

### CODE REVIEW

**Review by steverao — COMMENTED** [review]
Please run below command to fix CI problem: `./gradlew :javaagent:spotlessApply`

**Comment by trask** [comment]
@SylvainJuge the `license-check` job is failing, probably related to the changes?

**Comment by SylvainJuge** [comment]
> @SylvainJuge the `license-check` job is failing, probably related to the changes?

I've fixed the git check that failed as it assumed there was a single line change due to line with date. Should be good now.

File: javaagent/build.gradle.kts [hunk 1]
```diff
@@ -240,11 +242,30 @@ tasks {
...
+      val filePath = rootDir.toPath().resolve("licenses").resolve("licenses.md")
+      if (Files.exists(filePath)) {
+        val datePattern = Pattern.compile("^_[0-9]{4}-[0-9]{2}-[0-9]{2} [0-9]{2}:[0-9]{2}:[0-9]{2} .*_$")
+        val lines = Files.readAllLines(filePath).map {
+          if (datePattern.matcher(it).matches()) {
+            ""
+          } else {
+            it
+          }
+        }.toCollection(ArrayList<String>())
```
Comment (laurit) [inline javaagent/build.gradle.kts:257]: The line with date is always 4th line in the file, I'd just check that line.
It is unfortunate that
```
  class LicenseReportRenderer : InventoryMarkdownReportRenderer() {
    override fun printHeader() {
      output.appendText("## Dependency License Report\n")
    }
  }
  renderers = arrayOf(LicenseReportRenderer())
```
we can't use. Apparently extending `InventoryMarkdownReportRenderer`, that is written in groovy, from kotlin won't work correctly.

**Comment by laurit** [comment]
> Doing this means that the generateLicenseReport could be executed systematically to keep the license-report up-to-date when changes are made to dependencies, for example using a git pre-commit or pre-push hook.

I'm not sure whether we want to do this, but this change will definitely reduce the amount of merge conflicts when multiple PRs update the licenses file at the same time. Although these conflicts aren't hard to solve I think this PR is still useful. @trask do you have any objections to this?

**Comment by trask** [comment]
No objection to this PR from me.

**Review by laurit — APPROVED** [review]

**Pull request merged**